Customer and Comission Endpoints

CyberSturgeon opened the pull request:
+ middleware
+ simple get endpoints for customer
+ simple get endpoints for comission

CyberSturgeon committed:
wip

CyberSturgeon committed:
+ comission + middleware

Negodyaj commented on .gitignore:
```diff
@@ -363,3 +363,5 @@ MigrationBackup/
 /ReportingService.Presentanion/appsettings.json
 /ReportingService.Presentanion/appsettings.Development.json
+/ReportingService.Presentanion/appsettings.Dev.json
+/ReportingService.Presentanion/appsettings.Dev.json
```
а зачем дважды?)

CyberSturgeon replied:
убрал

Negodyaj commented on ReportingService.Application/Services/CustomerService.cs:
```diff
@@ -37,7 +37,7 @@ public async Task<CustomerModel> GetFullCustomerByIdAsync(Guid id)
 
         var accounts = await accountRepository.FindManyAsync(x => x.CustomerId == id);
-        if(!accounts.Any())
+        if (!accounts.Any())
```
бро, а давай ты в 36-й строке будешь тянуть кастомера сразу с аккаунтами?
.Include() в EF не просто так придумали)

Negodyaj commented on ReportingService.Persistence/Configuration/CustomerEntityConfiguration.cs:
```diff
@@ -18,8 +18,8 @@ internal static void ConfigureCustomer(this ModelBuilder builder)
-        builder.Entity<Customer>().HasIndex(x => x.Login)
-        .IsUnique();
+        //builder.Entity<Customer>().HasIndex(x => x.Login)
+        //.IsUnique();
```
если код не нужен, то не комменти, а удаляй

CyberSturgeon replied:
Там непонятки, у Юли нет логина, но в диалогах мы обсуждали что нужен логин, уберу, понял

CyberSturgeon replied:
убрал

Negodyaj commented on ReportingService.Persistence/Entities/Customer.cs:
```diff
@@ -8,7 +8,7 @@ public class Customer
     public string Phone { get; set; }
     public string Address { get; set; }
-    public string Login { get; set; }
+    //public string Login { get; set; }
```
логина а кастомер-сервисе нет
почта у нас вместо логина

Negodyaj commented on ReportingService.Presentanion/Configuration/ExceptionMiddleware.cs:
```diff
@@ -0,0 +1,52 @@
+    private async Task HandleExceptionAsync(HttpContext httpContext, Exception ex)
+    {
+        httpContext.Response.ContentType = "application/json";
+        httpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
```
ну вот тут точно не 404
и ты можешь вынести общие строки в отдельный метод?
если ещё пару обработчиков повесишь, будет оч много копипасты

CyberSturgeon replied:
Refactor middleware

Negodyaj commented on ReportingService.Presentanion/Controllers/ComissionController.cs:
```diff
@@ -0,0 +1,21 @@
+    IMapper mapper) : Controller
+{
+    [HttpGet("{id}")]
+    public async Task<ActionResult<ComissionResponse>> GetComissionByIdAsync([FromRoute] Guid id)
```
а почему именно этот эндпоинт?)
комиссии за период дат были бы полезнее

Negodyaj commented on ReportingService.Presentanion/Models/CustomerResponse.cs:
```diff
@@ -1,5 +1,20 @@
+    //public ICollection<AccountModel> Accounts { get; set; } = [];
+    public bool IsDeactivated { get; set; }
+    // DENORMALIZED
+    //public ICollection<TransactionModel> Transactions { get; set; } = [];
```
почисти от комментов

CyberSturgeon replied:
Висят ждут пока Сережин МР одобрится и там появятся TransactionResponse и AccountResponse, мастор

CyberSturgeon committed:
Change BaseRepo

+ include in FindAsync
fix tests for FindAsync
fix TestCases

CyberSturgeon committed:
fix GetFullCustomerById

CyberSturgeon committed:
Изменил метод поиска клиентов по фильтрам

CyberSturgeon committed:
fix bday search

CyberSturgeon committed:
fix tests

CyberSturgeon committed:
Не сменил ветку((

Тут интеграционные изменения для работы с CustomersService. Посмотри миграцию, я не стал её применять, она меня малям напугала

CyberSturgeon committed:
ready for rewiew

CyberSturgeon committed:
now ready

CyberSturgeon committed:
fix naming

CyberSturgeon committed:
ready

CyberSturgeon committed:
fix comission endpoint

Negodyaj committed:
Merge branch 'CustoMEME' into Features/GetCustomersAsync

Negodyaj committed:
Merge pull request #8 from CyberSturgeon/Features/GetCustomersAsync

Изменил метод поиска клиентов по фильтрам

CyberSturgeon committed:
Refactor middleware

+CustomException

Negodyaj commented on ReportingService.Application/Exceptions/CustomException.cs:
```diff
@@ -0,0 +1,8 @@
+
+public class CustomException(string message, HttpStatusCode code) : Exception(message)
+{
+    public readonly HttpStatusCode StatusCode = code;
```
HttpStatusCode не должен быть в проекте Application

CyberSturgeon committed:
find gitignore trouble

CyberSturgeon committed:
comission config

CyberSturgeon committed:
Refactor middleware again
